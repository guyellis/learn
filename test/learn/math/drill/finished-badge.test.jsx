
const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const FinishedBadge = require('../../../../src/learn/math/drill/finished-badge');

const muiTheme = getMuiTheme(lightBaseTheme);

describe('Finished Badge', () => {
  test('should be rendered on happy path', () => {
    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <FinishedBadge
          levelIndex={0}
          opIndexes={[0]}
          timePerQuestion={2.2}
          totalCorrectAnswers={10}
        />
      </MuiThemeProvider>);
    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });

  test('should be rendered if multiple opIndexes', () => {
    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <FinishedBadge
          levelIndex={0}
          opIndexes={[0, 1]}
          timePerQuestion={2.2}
          totalCorrectAnswers={10}
        />
      </MuiThemeProvider>);
    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });

  test('should not be rendered if correct answers under 10', () => {
    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <FinishedBadge
          levelIndex={0}
          opIndexes={[0]}
          timePerQuestion={2.2}
          totalCorrectAnswers={9}
        />
      </MuiThemeProvider>);
    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });
});
