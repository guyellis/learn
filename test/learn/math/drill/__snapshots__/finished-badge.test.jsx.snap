// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Finished Badge should be rendered if multiple opIndexes 1`] = `
<div
  style={Object {}}
>
  <span>
    <div
      aria-describedby={null}
      className="MuiAvatar-root MuiAvatar-colorDefault"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#9e9e9e",
        }
      }
      title="Mixed Silver"
    >
      M
    </div>
  </span>
  <span>
     Congratulations on getting a new Silver Badge for Level A Mixed Addition / Subtraction!
  </span>
</div>
`;

exports[`Finished Badge should be rendered on happy path 1`] = `
<div
  style={Object {}}
>
  <span>
    <div
      aria-describedby={null}
      className="MuiAvatar-root MuiAvatar-colorDefault"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#9e9e9e",
        }
      }
      title="+ Silver"
    >
      +
    </div>
  </span>
  <span>
     Congratulations on getting a new Silver Badge for Level A Addition!
  </span>
</div>
`;

exports[`Finished Badge should not be rendered if correct answers under 10 1`] = `
<div
  style={Object {}}
>
  You need 10 or more correct answers to earn a badge. On this 
test you got 9 answer(s) correct.
</div>
`;
